kube-runtime/src/reflector/object_ref.rs

```diff
@@ -44,6 +48,19 @@ pub struct ObjectRef<K: Resource> {
     /// assert_ne!(ObjectRef::<ConfigMap>::new("foo"), ObjectRef::new("foo").within("bar"));
     /// ```
     pub namespace: Option<String>,
+    /// Extra information about the object being referred to
+    ///
+    /// This is *not* considered when comparing objects, but may be used when converting to and from other representations,
+    /// such as [`OwnerReference`] or [`ObjectReference`].
+    #[derivative(Hash = "ignore", PartialEq = "ignore")]
+    pub extra: Extra,
```

Comment:
this seems like a sensible way to split this.

this might not have a benefit, but if we put a `#[serde(flatten)]` on `extra` then this is compatible with `ObjectReference` in terms of serialization

Reply:
That's a good point. Currently `ObjectRef` isn't `Serialize` anyway (and I don't think this is the place to change that), but that will probably not stay the case forever...